pkg/vm/ch.go

```diff
@@ -131,12 +93,16 @@ func (m *Machine) Run(ctx context.Context, socket, logs string) (pkg.MachineInfo
 
 		for _, nic := range m.Interfaces {
 			var typ InterfaceType
-			typ, _, err = nic.getType()
+			typ, idx, err := nic.getType()
```

Comment:
When you do Attach on the resource, you should create a tap device (not macvtap) which makes it easier to attach it to CH command line. The macvtap case was only used for public IPs specifically.